### DISCUSSION TITLE
Multiple segments with the same fetching information property

### DISCUSSION THREAD

**Discussion by accedo-robcc** [JanDeDobbeleer/oh-my-posh#5499] [Troubleshoot] [answered]
If I were to have multiple segments with the same fetching information (eg. `fetch_status`), is that less performant than having both on the same segment?

```json
{
  "type": "git",
  "style": "plain",
  "foreground": "p:blue",
  "properties": {
    "fetch_status": true
  },
  "template": "{{ if gt .Ahead 0 }} \u2191{{.Ahead}}{{end}}{{ if gt .Behind 0 }} \u2193{{.Behind}}{{end}} "
},
{
  "type": "git",
  "style": "plain",
  "foreground": "p:blue",
  "properties": {
    "fetch_status": true
  },
  "template": "{{ if gt .Working.Added 0 }}+{{ .Working.Added }} {{ end }}{{ if gt .Working.Modified 0 }}<p:git-modified>~{{ .Working.Modified }}</> {{ end }}{{ if gt .Working.Deleted 0 }}-{{ .Working.Deleted }} {{ end }}{{ if gt .Working.Untracked 0 }}?{{ .Working.Untracked }}{{ end }} "
}
```

**Comment by lewis-yeung** [comment]
The performance impact of a `git` segment is usually more significant when we're in a large-sized local repo.

**Reply by accedo-robcc** [reply]
but if I separate a git segment into two (with the same fetch flag), am I duplicating the performance needed? Or is the git status "cached" instead of being fetched twice

**Reply by JanDeDobbeleer** [reply, marked as answer]
@accedo-robcc it's fetched twice. Unless if you use [cross segment properties](https://ohmyposh.dev/docs/configuration/templates#cross-segment-template-properties) in a text segment for example.

**Reply by accedo-robcc** [reply]
good to know, and thanks for the link!
